aspnet/fundamentals/servers.rst

```diff
@@ -1,8 +1,274 @@
-.. include:: /../common/stub-topic.txt
+Servers
+=======
 
-|stub-icon| Servers
-===================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+ASP.NET 5 is completely decoupled from the web server environment that hosts the application. As released, ASP.NET 5 supports IIS and IIS Express, WebListener, and Kestrel web servers, which run on a variety of platforms. Developers and third party software vendors can create their own custom servers as well within which to host their ASP.NET 5 applications.
 
-.. _issue: https://github.com/aspnet/Docs/issues/59
+In this article:
+	- `Servers and commands`_
+	- `Supported features by server`_
+	- `IIS and IIS Express`_
+	- `WebListener`_
+	- `Kestrel`_
+	- `Choosing a server`_
+	- `Custom Servers`_
+	
+`Browse or download samples on GitHub <https://github.com/aspnet/Docs/tree/master/docs/fundamentals/servers/sample>`_.
+
+Servers and commands
+--------------------
+
+ASP.NET 5 is designed to decouple web applications from the underlying web server that hosts them. Traditionally, ASP.NET applications have been windows-only (unless `hosted via mono <http://www.mono-project.com/docs/web/aspnet/>`_) and hosted on the built-in web server for windows, Internet Information Server (IIS) (or a development server, like `IIS Express <http://www.iis.net/learn/extensions/introduction-to-iis-express/iis-express-overview>`_ or earlier development web servers). While IIS is still the recommended way to host production ASP.NET applications on Windows, the cross-platform nature of ASP.NET allows it to be hosted in any number of different web servers, on multiple operating systems.
+
+ASP.NET 5 ships with support for 3 different servers:
+
+- Microsoft.AspNet.Loader.IIS (Helios)
+- Microsoft.AspNet.Server.WebListener (WebListener)
+- Microsoft.AspNet.Server.Kestrel (Kestrel)
+
+ASP.NET 5 does not directly listen for requests, but instead relies on the HTTP server implementation to surface the request to the application as a set of `feature interfaces`_ composed into an HttpContext. Both Helios and WebListener are Windows-only; Kestrel is designed to run cross-platform. You can configure your application to be hosted by any or all of these servers by specifying commands in your ``project.json`` file. You can even specify an application entry point for your application, and run it as an executable (using ``dnx . run``) rather than hosting it in a separate process.
+
+I've configured the sample project for this article to support each of these hosting options as separate commands in ``project.json``:
+
+.. literalinclude:: servers/sample/ServersDemo/src/ServersDemo/project.json
+	:lines: 1-17
+	:emphasize-lines: 12-14
+	:linenos:
+	:language: javascript
+	:caption: project.json (truncated)
+
+The ``run`` command will execute the application via its ``void main()`` method defined in ``program.cs``. In this case, this has been set up to configure and start an instance of ``WebListener``.
```

Comment:
Do you mean in the Hosting article #58 or the Custom Servers section in this article? I agree it's not extremely common but it is used in MusicStore, so I figured it was worth noting here for completeness' sake.